src/main/scala/org/apache/daffodil/DaffodilPlugin.scala

```diff
@@ -458,7 +461,8 @@ object DaffodilPlugin extends AutoPlugin {
             // where API functions to use. This is the mapping for which Daffodil API should be
             // used for a particular "internal API"
             val daffodilInternalApiVersionMapping = Map(
-              ">3.8.0" -> "2",
+              ">=4.0.0" -> "3",
+              "<=3.11.0 >3.8.0" -> "2",
```

Comment:
Following the convention of the other ranges in this plugin, this would be `>=3.9.0 <=3.11.0 -> "2"`